File: src/main/java/com/github/twitch4j/streamlabs4j/api/domain/StreamlabsUser.java
package com.github.twitch4j.streamlabs4j.api.domain;

import com.fasterxml.jackson.annotation.JsonIgnoreProperties;
import com.fasterxml.jackson.databind.PropertyNamingStrategy;
import com.fasterxml.jackson.databind.annotation.JsonNaming;
import lombok.AccessLevel;
import lombok.Data;
import lombok.NoArgsConstructor;
import lombok.Setter;

import java.util.Optional;

/**
 * Streamlabs User
 */
@Data
@Setter(AccessLevel.PRIVATE)
@NoArgsConstructor
@JsonNaming(PropertyNamingStrategy.SnakeCaseStrategy.class)
@JsonIgnoreProperties(ignoreUnknown = true)
public class StreamlabsUser {

    // Streamlabs User Information
    private StreamlabsUserInformation streamlabs;

    // Twitch User Information
    private Optional<TwitchUserInformation> twitch;

    // Youtube User Information
    private Optional<YoutubeUserInformation> youtube;

    // Mixer User Information
    private Optional<MixerUserInformation> mixer;

    // Facebook User Information
    private Optional<FacebookUserInformation> facebook;

    @Data
    @Setter(AccessLevel.PRIVATE)
    @NoArgsConstructor
    @JsonNaming(PropertyNamingStrategy.SnakeCaseStrategy.class)
    @JsonIgnoreProperties(ignoreUnknown = true)
    public static class StreamlabsUserInformation {

        // UserId
        private Long id;

        // DisplayName
        private String displayName;

    }

    @Data
    @Setter(AccessLevel.PRIVATE)
    @NoArgsConstructor
    @JsonNaming(PropertyNamingStrategy.SnakeCaseStrategy.class)
    @JsonIgnoreProperties(ignoreUnknown = true)
    public static class TwitchUserInformation {

        // UserId
        private Long id;

        // Name
        private String name;

        // DisplayName
        private String displayName;

    }

    @Data
    @Setter(AccessLevel.PRIVATE)
    @NoArgsConstructor
    @JsonNaming(PropertyNamingStrategy.SnakeCaseStrategy.class)
    @JsonIgnoreProperties(ignoreUnknown = true)
    public static class YoutubeUserInformation {

        // UserId
        private String id;

        // Title
        private String title;

    }

    @Data
    @Setter(AccessLevel.PRIVATE)
    @NoArgsConstructor
    @JsonNaming(PropertyNamingStrategy.SnakeCaseStrategy.class)
    @JsonIgnoreProperties(ignoreUnknown = true)
    public static class MixerUserInformation {

        // UserId
        private Long id;

        // Name
        private String name;

    }

    @Data
    @Setter(AccessLevel.PRIVATE)
    @NoArgsConstructor
    @JsonNaming(PropertyNamingStrategy.SnakeCaseStrategy.class)
    @JsonIgnoreProperties(ignoreUnknown = true)
    public static class FacebookUserInformation {

        // UserId
        private Long id;

        // Name
        private String name;

    }

}

File: src/main/kotlin/com/github/twitch4j/streamlabs4j/api/domain/StreamlabsUser.kt
package com.github.twitch4j.streamlabs4j.api.domain

import com.fasterxml.jackson.annotation.JsonIgnoreProperties
import com.fasterxml.jackson.databind.PropertyNamingStrategy.SnakeCaseStrategy
import com.fasterxml.jackson.databind.annotation.JsonNaming

/** Streamlabs User */
@JsonNaming(SnakeCaseStrategy::class)
@JsonIgnoreProperties(ignoreUnknown = true)
data class StreamlabsUser(
    val streamlabs: StreamlabsUserInformation,
    val youtube: YoutubeUserInformation?,
    val twitch: TwitchUserInformation?,
    val mixer: MixerUserInformation?,
    val facebook: FacebookUserInformation?
) {

    /** Streamlabs information */
    @JsonNaming(SnakeCaseStrategy::class)
    @JsonIgnoreProperties(ignoreUnknown = true)
    data class StreamlabsUserInformation(
        val id: Long,
        val displayName: String
    )

    /** Twitch information */
    @JsonNaming(SnakeCaseStrategy::class)
    @JsonIgnoreProperties(ignoreUnknown = true)
    data class TwitchUserInformation(
        val id: Long,
        val name: String,
        val displayName: String
    )

    /** YouTube information */
    @JsonNaming(SnakeCaseStrategy::class)
    @JsonIgnoreProperties(ignoreUnknown = true)
    data class YoutubeUserInformation(val id: Long, val title: String)

    /** Mixer information */
    @JsonNaming(SnakeCaseStrategy::class)
    @JsonIgnoreProperties(ignoreUnknown = true)
    data class MixerUserInformation(val id: Long, val name: String)

    /** Facebook information */
    @JsonNaming(SnakeCaseStrategy::class)
    @JsonIgnoreProperties(ignoreUnknown = true)
    data class FacebookUserInformation(val id: Long, val name: String)

}
